tests/composite_pk/test_update.py

```diff
@@ -110,6 +137,14 @@ def test_update_or_create_user(self):
                 self.assertEqual(user.email, fields["defaults"]["email"])
                 self.assertEqual(count, User.objects.count())
 
+    def test_update_or_create_with_pre_save_pk_field(self):
+        t = TimeStamped.objects.create(id=1)
+        t, created = TimeStamped.objects.update_or_create(
+            pk=t.pk, defaults={"text": "new text"}
+        )
+        self.assertIs(created, False)
+        self.assertIs(t.text, "new text")
```

Comment:
what we should be testing here is that an auto_now field wasn't updated if part of a composite primary key

Reply:
I believe this is right. Without the patch `update_or_create` crashes because it passes `created` to `update_fields` but changes we're also made to deny doing so. See the `test_update_fields_pk_field` test.

```suggestion
        self.asserEqua(t.text, "new text")
```

`assertEqual` should be used as two string sharing the same address is a CPython implementation detail.

Reply:
hmm okay I see, fair enough